packages/grafana-ui/src/components/AutoSaveField/AutoSaveField.story.tsx

```diff
@@ -0,0 +1,255 @@
+import { Story, Meta } from '@storybook/react';
+import React, { useState } from 'react';
+
+import { withCenteredStory } from '../../utils/storybook/withCenteredStory';
+import { Checkbox } from '../Forms/Checkbox';
+import { RadioButtonGroup } from '../Forms/RadioButtonGroup/RadioButtonGroup';
+import { Input } from '../Input/Input';
+import { Select } from '../Select/Select';
+import { Switch } from '../Switch/Switch';
+import { TextArea } from '../TextArea/TextArea';
+
+import { AutoSaveField } from './AutoSaveField';
+import mdx from './AutoSaveField.mdx';
+
+const meta: Meta = {
+  title: 'AutoSaveField',
+  component: AutoSaveField,
+  decorators: [withCenteredStory],
+  parameters: {
+    docs: {
+      page: mdx,
+    },
+    controls: {
+      exclude: [
+        'prefix',
+        'width',
+        'loading',
+        'suffix',
+        'addonBefore',
+        'addonAfter',
+        'onFinishChange',
+        'invalid',
+        'weekPickerValue',
+      ],
+    },
+  },
+  argTypes: {
+    saveErrorMessage: { control: 'text' },
+    label: { control: 'text' },
+    required: {
+      control: { type: 'select', options: [true, false] },
+    },
+  },
+  args: {
+    saveErrorMessage: 'This is a custom error message',
+    required: false,
+  },
+};
+
+export default meta;
+
+const getSuccess = () => {
+  return new Promise<void>((resolve) => {
+    setTimeout(resolve, 1000);
+  });
+};
+const getError = () => {
+  return new Promise<void>((resolve, reject) => {
+    reject();
+  });
+};
+const themeOptions = [
+  { value: '', label: 'Default' },
+  { value: 'dark', label: 'Dark' },
+  { value: 'light', label: 'Light' },
+  { value: 'system', label: 'System' },
+];
+enum ThemeSwitchOption {
+  Light = 'light',
+  Dark = 'dark',
+}
+
+export const AutoSaveFieldError: Story = (args) => {
```

Comment:
I would add one more story that only has a single component in it (probably the Input) and call it something like "Basic", just so we have a simpler story. Then you can call the other ones "All components success"/"All components error", what do you think?